src/ceph-volume/ceph_volume/tests/test_terminal.py

```diff
@@ -66,3 +71,50 @@ def test_subcommand_found_and_dispatched_with_errors(self):
         with pytest.raises(SystemExit) as error:
             terminal.dispatch({'sub': BadSubCommand}, argv=['sub'])
         assert str(error.value) == '100'
+
+
+@pytest.fixture
+def stream():
+    def make_stream(buffer, encoding):
+        # mock a stdout with given encoding
+        if sys.version_info >= (3, 0):
+            stdout = sys.stdout
+            stream = io.TextIOWrapper(buffer,
+                                      encoding=encoding,
+                                      errors=stdout.errors,
+                                      newline=stdout.newlines,
+                                      line_buffering=stdout.line_buffering)
+        else:
+            stream = codecs.getwriter(encoding)(buffer)
+            # StreamWriter does not have encoding attached to it, it will ask
+            # the inner buffer for "encoding" attribute in this case
+            stream.encoding = encoding
+        return stream
+    return make_stream
+
+
+class TestWriteUnicode(object):
+
+    def setup(self):
+        self.octpus_and_squid_en = u'octpus and squid'
+        octpus_and_squid_zh = u'章鱼和鱿鱼'
+        self.message = self.octpus_and_squid_en + octpus_and_squid_zh
+
+    def test_stdout_writer(self, capsys):
+        # should work with whatever stdout is
+        terminal.stdout(self.message)
+        out, _ = capsys.readouterr()
+        assert self.octpus_and_squid_en in out
+
+    @pytest.mark.parametrize('encoding', ['ascii', 'utf8'])
+    def test_writer(self, encoding, stream, monkeypatch, capsys):
+        buffer = io.BytesIO()
+        # should keep writer alive
+        with capsys.disabled():
+            # we want to have access to the sys.stdout's attributes in
+            # make_stream(), not the ones of pytest.capture.EncodedFile
+            writer = stream(buffer, encoding)
```

Comment:
also, instead of passing a temporary instance of writer. i found that i need to keep it alive during the test.